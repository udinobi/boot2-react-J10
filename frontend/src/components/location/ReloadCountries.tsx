import { Button } from "antd"
import React from "react"
import { connect } from "react-redux"

import {
    CountriesProps,
    mapCountryDispatchToProps
} from "../../store/location/country/actions"

export const ReloadCountries = (props: CountriesProps) => {
    {/* tslint:disable:jsx-boolean-value */}
    return <Button
        data-tip data-for="ReloadCountries"
        icon="reload"
        onClick={props.loadCountries}
        shape="circle"
        type="primary"
    />
}

export default connect(undefined, mapCountryDispatchToProps)(ReloadCountries);
