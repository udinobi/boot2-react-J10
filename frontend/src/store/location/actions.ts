import { ActionCreator } from "redux"

import { Location } from "./types"

import { LocationChangedAction } from "./types"

export const changeCurrentLocation: ActionCreator<LocationChangedAction> = (location: Location) => ({
    payload: {
        location,
    },
    type: "location/LOCATION_CHANGED"
})
