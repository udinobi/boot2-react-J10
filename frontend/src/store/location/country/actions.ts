import axios, { AxiosResponse } from "axios"
import { ActionCreator, Dispatch } from "redux"

import {
    CountriesFailureAction, CountriesLoadingAction, CountriesRetrievedAction,
    Country, CountryActions, CountryActionType,
    CountrySelectedAction, CountryState
} from "./types"

// CountriesFailureAction --------------------------------------------------------------------------

const countriesFailure: ActionCreator<CountriesFailureAction> = () => ({
    type: CountryActionType.COUNTRIES_FAILURE
})

// CountriesLoadAction -----------------------------------------------------------------------------

const countriesLoading: ActionCreator<CountriesLoadingAction> = () => ({
    type: CountryActionType.COUNTRIES_LOADING
})

// Not checking yet, at this stage in our CSR app, if these properties are set or not.
// In a future SSR rendering feature branch we can also consider to implement logging
// to the server-side RT.
const url =
    `${process.env.REACT_APP_SERVICE_URL}${process.env.REACT_APP_LOAD_COUNTRIES_PATH}`

export const loadCountries = () =>
    async (dispatch: Dispatch<CountryActions>) => {
        dispatch(countriesLoading())

        try {
            const response: AxiosResponse<CountryState> = await axios.get(url)
            dispatch(countriesRetrieved(response.data))
        } catch (error) {
            dispatch(countriesFailure())
        }
    }
        
// CountriesRetrievedAction ------------------------------------------------------------------------

const countriesRetrieved: ActionCreator<CountriesRetrievedAction> = (countries: Country[]) => ({
    payload: { countries },
    type: CountryActionType.COUNTRIES_RETRIEVED,
})

// CountrySelectedAction ---------------------------------------------------------------------------

export const countrySelectedAction: ActionCreator<CountrySelectedAction> = (country: Country) => ({
    payload: { country },
    type: CountryActionType.COUNTRY_SELECTED
})

export const countrySelected = (country: Country) =>
    (dispatch: Dispatch<CountrySelectedAction>) => dispatch(countrySelectedAction(country))

// Action mapping ----------------------------------------------------------------------------------

export interface CountriesProps {
    countries: Country[]
    countrySelected: typeof countrySelected
    loadCountries: typeof loadCountries
}

export const mapCountryDispatchToProps = {
    countrySelected,
    loadCountries
}
    