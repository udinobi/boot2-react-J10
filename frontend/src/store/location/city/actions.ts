import { ActionCreator, Dispatch } from "redux"

import {
    CityActionType,
    Location, LocationSelectedAction,
    SuggestionsFailureAction, SuggestionsLookupAction,
    SuggestionsResetAction, SuggestionsRetrievedAction
} from "./types"

import { Country } from "../country/types"

// LocationSelectedAction --------------------------------------------------------------------------

const locationSelectedAction: ActionCreator<LocationSelectedAction> = (location: Location) => ({
    payload: { location },
    type: CityActionType.LOCATION_SELECTED
})

export const locationSelected = (location: Location) =>
    (dispatch: Dispatch<LocationSelectedAction | SuggestionsResetAction>) => {
        dispatch(suggestionsReset())
        dispatch(locationSelectedAction(location))
    }

// SuggestionsFailureAction ------------------------------------------------------------------------

export const suggestionsFailure: ActionCreator<SuggestionsFailureAction> = () => ({
    type: CityActionType.SUGGESTIONS_FAILURE
})

// SuggestionsLookupAction -------------------------------------------------------------------------

export const suggestionsLookup: ActionCreator<SuggestionsLookupAction> = (country: Country, locationTerm: string) => ({
    payload: { country, locationTerm },
    type: CityActionType.SUGGESTIONS_LOOKUP
})

// SuggestionsResetAction ------------------------------------------------------------------------

const suggestionsReset: ActionCreator<SuggestionsResetAction> = () => ({
    type: CityActionType.SUGGESTIONS_RESET
})

// SuggestionsRetrievedAction ----------------------------------------------------------------------

export const suggestionsRetrieved: ActionCreator<SuggestionsRetrievedAction> = (country: Country, suggestions: Location[]) => ({
    payload: { country, suggestions },
    type: CityActionType.SUGGESTIONS_RETRIEVED
})

// Action mapping ----------------------------------------------------------------------------------

export interface CityDispatchProps {
    locationSelected: typeof locationSelected
    suggestionsLookup: typeof suggestionsLookup
    suggestionsReset: typeof suggestionsReset
}

export const mapCityDispatchToProps = {
    locationSelected,
    suggestionsLookup,
    suggestionsReset
}
