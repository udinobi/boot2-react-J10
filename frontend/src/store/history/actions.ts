import { ActionCreator, Dispatch } from "redux"

import { Location } from "../location/city/types"

import {
    HistoryActionType, HistoryItem, LocationReloadAction, LocationRemoveAction
} from './types'

// LocationReloadAction ----------------------------------------------------------------------------

const locationReloadAction: ActionCreator<LocationReloadAction> = (history: HistoryItem) => ({
    payload: { history },
    type: HistoryActionType.LOCATION_RELOAD
})

export const reloadLocation = (history: HistoryItem) =>
    (dispatch: Dispatch<LocationReloadAction>) => dispatch(locationReloadAction(history))

// LocationRemoveAction ----------------------------------------------------------------------------

const locationRemoveAction: ActionCreator<LocationRemoveAction> = (location: Location) => ({
    payload: { location },
    type: HistoryActionType.LOCATION_REMOVE
})

const removeLocation = (location: Location) =>
    (dispatch: Dispatch<LocationRemoveAction>) => dispatch(locationRemoveAction(location))

// Action mapping ----------------------------------------------------------------------------------

export interface HistoryDispatchProps {
    reloadLocation: typeof reloadLocation
    removeLocation: typeof removeLocation
}

export const mapHistoryDispatchToProps = {
    reloadLocation,
    removeLocation
}
