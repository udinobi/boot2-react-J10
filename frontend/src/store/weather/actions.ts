import { ActionCreator } from "redux"

import {
    WeatherActionType, WeatherData, WeatherDataFailureAction,
    WeatherDataRetrieveAction, WeatherDataRetrievedAction } from "./types"

import { Location } from "../location/city/types"

// WeatherDataFailureAction ------------------------------------------------------------------------

export const weatherDataFailure: ActionCreator<WeatherDataFailureAction> = () => ({
    type: WeatherActionType.WEATHER_DATA_FAILURE
})

// WeatherDataRetrieveAction -----------------------------------------------------------------------

export const weatherDataRetrieve: ActionCreator<WeatherDataRetrieveAction> =
    (location: Location, owmApiKey: string) => ({
        payload: { location, owmApiKey },
        type: WeatherActionType.WEATHER_DATA_RETRIEVE
    })

// WeatherDataRetrievedAction ----------------------------------------------------------------------

export const weatherDataRetrieved: ActionCreator<WeatherDataRetrievedAction> =
    (location: Location, weatherData: WeatherData) => ({
        payload: { location, weatherData },
        type: WeatherActionType.WEATHER_DATA_RETRIEVED
    })

// Action mapping ----------------------------------------------------------------------------------

export interface WeatherDispatchProps {
    weatherDataRetrieve: typeof weatherDataRetrieve
}

export const mapWeatherDispatchToProps = {
    weatherDataRetrieve
}
