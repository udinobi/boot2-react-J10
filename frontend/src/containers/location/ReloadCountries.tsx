import { Button } from "antd"
import React from "react"
import { connect } from "react-redux"

import { updateCountries } from "../../store/location/actions"

interface DispatchProps {
    updateCountries: typeof updateCountries
}

const ReloadCountries = (props: DispatchProps) =>
    <Button icon="reload" onClick={props.updateCountries} shape="circle" />

const mapDispatchToProps = {
    updateCountries
}

export default connect(undefined, mapDispatchToProps)(ReloadCountries);
