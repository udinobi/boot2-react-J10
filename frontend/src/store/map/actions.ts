import { reloadLocation } from '../history/actions'
import { locationSelected } from '../location/cityActions'

export interface MapDispatchProps {
    locationSelected: typeof locationSelected
    reloadLocation: typeof reloadLocation
}

export const mapMapDispatchToProps = {
    locationSelected,
    reloadLocation
}
