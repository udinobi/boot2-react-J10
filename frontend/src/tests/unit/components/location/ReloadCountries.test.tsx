import React from "react"
import * as Renderer from 'react-test-renderer'

import { ReloadCountries } from "../../../../components/location/ReloadCountries"

import { CountriesProps } from "../../../../store/location/country/actions"

describe("--> Snapshot for Components > location > 'ReloadCountries'", () => {
    let props: CountriesProps

    beforeEach(() => {
        props = {
            countries: [],
            countrySelected: jest.fn(),
            loadCountries: jest.fn
        }
    })

    it("matches", () => {
        const component = Renderer.create(<ReloadCountries {...props} />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
