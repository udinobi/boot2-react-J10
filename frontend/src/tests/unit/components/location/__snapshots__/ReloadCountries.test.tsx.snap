// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--> Snapshot for Components > location > 'ReloadCountries' matches 1`] = `
<button
  className="ant-btn ant-btn-primary ant-btn-circle ant-btn-icon-only"
  data-for="ReloadCountries"
  data-tip={true}
  onClick={[Function]}
  type="button"
>
  <i
    className="anticon anticon-reload"
  />
</button>
`;
